### PR TITLE
Modifies BlowfishEngine code to more closely match the original.

### PR DESCRIPTION
**Pull request by tomsons26**

### CODE REVIEW

File: src/game/crypto/blowfish.cpp [hunk 1]
```diff
@@ -1092,47 +175,62 @@ void BlowfishEngine::Submit_Key(void *key, int length)
...
-            Sub_Key_Encrypt(m_sBox + i, m_sBox + i + 2);
+        
+        uint32_t left = 0;
+        uint32_t right = 0;
+
+        uint32_t *enc = m_pBoxEnc;
+        uint32_t *dec = &m_pBoxDec[BF_ROUNDS + 1];
+        for (int i = 0; i < BF_ROUNDS + 2; i += 2) {
+            Sub_Key_Encrypt(left, right);
+            /*
```
Comment (OmniBlade) [inline src/game/crypto/blowfish.cpp]: Remove comments like this from final code.

File: src/game/crypto/blowfish.cpp [hunk 2]
```diff
@@ -1141,111 +239,96 @@ void BlowfishEngine::Submit_Key(void *key, int length)
...
+        pboxa = pbox + 1;
+
+        right ^= *pboxa ^ (m_sBox[3][GETBYTE(xored, 0)]
+                     + (m_sBox[2][GETBYTE(xored, 1)] 
+                     ^ (m_sBox[1][GETBYTE(xored, 2)]
+                     + m_sBox[0][GETBYTE(xored, 3)])));
 
-    right ^= p[BF_ROUNDS + 1];
+        pbox = pboxa + 1;
+        /*
```
Comment (OmniBlade) [inline src/game/crypto/blowfish.cpp]: Again remove comments like this.

File: src/game/crypto/blowfish.cpp [hunk 3]
```diff
@@ -1254,26 +337,35 @@ void BlowfishEngine::Process_Block(void *input, void *output, uint32_t *pbox)
...
+
+        uint32_t xored = m_pBoxEnc[i] ^ ldata;
+
+        rdata ^= m_pBoxEnc[i + 1]
+            ^ (m_sBox[3][GETBYTE(xored, 0)]
+            + (m_sBox[2][GETBYTE(xored, 1)] 
+            ^ (m_sBox[1][GETBYTE(xored, 2)]
+            + m_sBox[0][GETBYTE(xored, 3)])));
+
+        /*
```
Comment (OmniBlade) [inline src/game/crypto/blowfish.cpp]: Remove.

**Review by OmniBlade — CHANGES_REQUESTED** [review]
If you are confident the code works after doing some testing, then remove the speculative comments and we will merge.

**Review by ghost — DISMISSED** [review]

**Review by ghost — DISMISSED** [review]

File: src/game/crypto/blowfish.cpp [hunk 4]
```diff
@@ -20,1057 +21,137 @@
 #include <captainslog.h>
 #include <string.h>
 
+// clang-format off
 const uint32_t BlowfishEngine::s_Blowfish_p_init[BF_ROUNDS + 2] = {
```
Comment (OmniBlade) [inline src/game/crypto/blowfish.cpp]: While we are messing with this code, change the name to s_PInit, I think having Blowfish in the name is overkill as its scoped in the class namespace anyhow.

File: src/game/crypto/blowfish.cpp [hunk 5]
```diff
@@ -20,1057 +21,137 @@
...
-        0xc208e69f,
-        0xb74e6132,
-        0xce77e25b,
-        0x578fdfe3,
-        0x3ac372e6 } };
+    0x243f6a88, 0x85a308d3, 0x13198a2e, 0x03707344, 0xa4093822, 0x299f31d0, 0x082efa98, 0xec4e6c89, 0x452821e6,
+    0x38d01377, 0xbe5466cf, 0x34e90c6c, 0xc0ac29b7, 0xc97c50dd, 0x3f84d5b5, 0xb5470917, 0x9216d5d9, 0x8979fb1b
+};
+
+const uint32_t BlowfishEngine::s_Blowfish_s_init[4][256] = {
```
Comment (OmniBlade) [inline src/game/crypto/blowfish.cpp]: Same with this one, s_SInit.

**Review by OmniBlade — CHANGES_REQUESTED** [review]

File: src/game/crypto/blowfish.cpp [hunk 6]
```diff
@@ -20,1062 +21,142 @@
...
+        0xf6fb2299, 0xf523f357, 0xa6327623, 0x93a83531, 0x56cccd02, 0xacf08162, 0x5a75ebb5, 0x6e163697, 0x88d273cc,
+        0xde966292, 0x81b949d0, 0x4c50901b, 0x71c65614, 0xe6c6c7bd, 0x327a140a, 0x45e1d006, 0xc3f27b9a, 0xc9aa53fd,
+        0x62a80f00, 0xbb25bfe2, 0x35bdd2f6, 0x71126905, 0xb2040222, 0xb6cbcf7c, 0xcd769c2b, 0x53113ec0, 0x1640e3d3,
+        0x38abbd60, 0x2547adf0, 0xba38209c, 0xf746ce76, 0x77afa1c5, 0x20756060, 0x85cbfe4e, 0x8ae88dd8, 0x7aaaf9b0,
+        0x4cf9aa7e, 0x1948c25c, 0x02fb8a8c, 0x01c36ae4, 0xd6ebe1f9, 0x90d4f869, 0xa65cdea0, 0x3f09252d, 0xc208e69f,
+        0xb74e6132, 0xce77e25b, 0x578fdfe3, 0x3ac372e6 }
+};
+// clang-format on
+
+BlowfishEngine::BlowfishEngine(void *key, int length) : m_HasKey(false)
```
Comment (OmniBlade) [inline src/game/crypto/blowfish.cpp:157]: If this is going to just be a default ctor, then remove the parameters as they are not used and thus not required.

**Review by OmniBlade — CHANGES_REQUESTED** [review]

**Review by OmniBlade — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**